addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiDeviceBaseHandler.java

```diff
@@ -123,13 +123,17 @@ public String getItemId() {
     void parseCommand(String command, JsonObject data) {
         if (command.equals("report")) {
             parseReport(data);
-        } else if (command.equals("heartbeat") || command.equals("read_ack") || command.equals("write_ack")) {
+        } else if (command.equals("heartbeat")) {
+            parseHeartbeat(data);
+        } else if (command.equals("read_ack") || command.equals("write_ack")) {
```

Comment:
In XiaomiSensorBaseHandler: we then could just call ```parseHeartbeat()``` from within ```parseReadAck()```
And in XiaomiActorGatewayHandler: we then could just call ```parseHeartbeat()``` from within ```parseReadAck()``` and ```parseWriteAck()```